test/unit/transaction_builder_test.js

```diff
@@ -260,7 +260,63 @@ describe('TransactionBuilder', function() {
       done();
     });
   });
-
+  describe('timebounds', function(){
+    it('requires maxTime', function(){
+      let source = new StellarBase.Account(
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ',
+        '0'
+      );
+      expect(() => {
+        new StellarBase.TransactionBuilder(source, {
+          timebounds: {
+            minTime: '0'
+          },
+          fee: 100
+        }).build();
+      }).to.throw('TimeBounds has to be set or you must call setTimeout(TimeoutInfinite).');
+    });
+    it('requires minTime', function(){
+      let source = new StellarBase.Account(
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ',
+        '0'
+      );
+      expect(() => {
+        new StellarBase.TransactionBuilder(source, {
+          timebounds: {
+            maxTime: '10'
+          },
+          fee: 100
+        }).build();
+      }).to.throw('TimeBounds has to be set or you must call setTimeout(TimeoutInfinite).');
+    });
+    it('works with timebounds defined', function(){
+      let source = new StellarBase.Account(
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ',
+        '0'
+      );
+      expect(() => {
+        new StellarBase.TransactionBuilder(source, {
+          timebounds: {
+            minTime: '1',
+            maxTime: '10'
+          },
+          fee: 100
+        }).build();
+      }).to.not.throw('TimeBounds has to be set or you must call setTimeout(TimeoutInfinite).');
```

Comment:
is it possible to assert that it doesn't throw any exception instead of testing for a specific exception?

Reply:
good catch! no exception should be thrown.